### PR TITLE
Secret documentation update:

### PR DESCRIPTION
**Pull request by protatremy**
- Interpolate secrets
- obfuscate secret
- secret in a service

## Contributor Checklist:
* [x] I have updated the appropriate documentation

### CODE REVIEW

**Comment by tardyp** [comment]
@seankelly can you double check the anglish please

File: master/docs/developer/secrets.rst [hunk 1]
```diff
@@ -68,8 +68,67 @@ In the master configuration, the provider is instantiated through a Buildbot ser
...
+When the service is loaded during the Buildbot reconfigService function, secrets are rendered and the value are updated.
+Everywhere the variable with the secret name (`foo` or `other` in the example) is used, the variable is replaced by the secret value.
+
+  .. code-block:: python
+
+      class MyService(object):
+        secrets = ['foo', 'other']
+
+        def returnSecretValue(self, secretkey):
+          return getatr(self, secretkey)
```
Comment (tardyp) [inline master/docs/developer/secrets.rst]: getattr

File: master/docs/developer/secrets.rst [hunk 2]
```diff
@@ -68,8 +68,67 @@ In the master configuration, the provider is instantiated through a Buildbot ser
...
+
+      myService = MyService()
+
+After a Buildbot reconfigService:
+
+  .. code-block:: python
+
+      print("myService returns secret value:", myService.returnSecretValue("foo"))
```
Comment (tardyp) [inline master/docs/developer/secrets.rst]: why no just myService.foo ?

File: master/docs/developer/secrets.rst [hunk 3]
```diff
@@ -68,8 +68,67 @@ In the master configuration, the provider is instantiated through a Buildbot ser
...
+Secrets are used like renderables in a service and are rendered during the configuration load.
+
+.. code-block:: python
+
+    class MyService(BuildbotService):
+      secrets = ['foo', 'other']
+
+``secrets`` is a list containing all the secrets key used in the class.
+When the service is loaded during the Buildbot reconfigService function, secrets are rendered and the value are updated.
+Everywhere the variable with the secret name (`foo` or `other` in the example) is used, the variable is replaced by the secret value.
```
Comment (tardyp) [inline master/docs/developer/secrets.rst]: [to add] this is similar to the "renderable" annotation, but will only works for BuildbotServices, and will only interpolate secrets. Other renderable can still be held in the service as attributes, and rendered dynamically at a later time.

File: master/docs/developer/secrets.rst [hunk 4]
```diff
@@ -68,8 +68,67 @@ In the master configuration, the provider is instantiated through a Buildbot ser
...
 
+Interpolate secret
+``````````````````
+
+.. code-block:: python
+
+    text = Interpolate("some text and %(secret:foo)s")
+
+Secret keys are replaced in a string by the secret value using the class Interpolate and the keyword secret.
+The secret is searched across the provider defined in the master configuration.
```
Comment (seankelly) [inline master/docs/developer/secrets.rst]: provider -> providers.

File: master/docs/developer/secrets.rst [hunk 5]
```diff
@@ -68,8 +68,67 @@ In the master configuration, the provider is instantiated through a Buildbot ser
...
+    cleantext = self.build.build_status.properties.cleanupTextFromSecrets(rendered)
+
+Secrets don't have to be visible to the normal user via logs and thus are transmitted directly to the workers.
+Secrets are rendered and can arrive anywhere in the logs.
+The function ``cleanupTextFromSecrets`` defined in the class Properties helps to replace the secret value by the key value.
+
+.. code-block:: python
+
+    print("the example value is:%s" % (cleantext))
+    >> the example value is: <secret>
```
Comment (seankelly) [inline master/docs/developer/secrets.rst]: The last sentence in the previous paragraph says the secret will be replaced by the key so I expected something like `foo` or `<foo>`.

File: master/docs/developer/secrets.rst [hunk 6]
```diff
@@ -68,8 +68,67 @@ In the master configuration, the provider is instantiated through a Buildbot ser
...
+
+Secret is rendered and it is recorded in a dictionary, named ``_used_secrets``, where the key is the secret value and the value the secret key.
+Therefore anywhere logs are written having content with secrets, secret are replaced by the key value containing in ``_used_secrets``.
```
Comment (seankelly) [inline master/docs/developer/secrets.rst]: Replace everything after the comma with:
> the secrets are replaced by the value from \`\`_used_secrets\`\`.

Note: this phrasing still isn't quite what I want so I may have a better version later.

File: master/docs/developer/secrets.rst [hunk 7]
```diff
@@ -68,8 +68,67 @@ In the master configuration, the provider is instantiated through a Buildbot ser
...
+
+How to use a secret in a BuildbotService
+````````````````````````````````````````
+
+Services configurations are loaded during a Buildbot start or modified during a Buildbot restart.
```
Comment (seankelly) [inline master/docs/developer/secrets.rst]: Services -> Service.

File: master/docs/developer/secrets.rst [hunk 8]
```diff
@@ -68,8 +68,67 @@ In the master configuration, the provider is instantiated through a Buildbot ser
...
+Secrets are used like renderables in a service and are rendered during the configuration load.
+
+.. code-block:: python
+
+    class MyService(BuildbotService):
+      secrets = ['foo', 'other']
+
+``secrets`` is a list containing all the secrets key used in the class.
```
Comment (seankelly) [inline master/docs/developer/secrets.rst]: secrets key -> secret keys.

Though, by "keys", do you mean keys to look up as a secret or parameters for the service?

File: master/docs/developer/secrets.rst [hunk 9]
```diff
@@ -68,8 +68,67 @@ In the master configuration, the provider is instantiated through a Buildbot ser
...
+When the service is loaded during the Buildbot reconfigService function, secrets are rendered and the value are updated.
```
Comment (seankelly) [inline master/docs/developer/secrets.rst]: value -> values.

File: master/docs/developer/secrets.rst (hunk 3 above)
Comment (protatremy) [inline master/docs/developer/secrets.rst]: ok

File: master/docs/developer/secrets.rst (hunk 1 above)
Comment (protatremy) [inline master/docs/developer/secrets.rst]: ok

File: master/docs/developer/secrets.rst (hunk 2 above)
Comment (protatremy) [inline master/docs/developer/secrets.rst]: ok

File: master/docs/developer/secrets.rst (hunk 4 above)
Comment (protatremy) [inline master/docs/developer/secrets.rst]: ok

File: master/docs/developer/secrets.rst (hunk 5 above)
Comment (protatremy) [inline master/docs/developer/secrets.rst]: ok

File: master/docs/developer/secrets.rst (hunk 6 above)
Comment (protatremy) [inline master/docs/developer/secrets.rst]: ok

File: master/docs/developer/secrets.rst (hunk 7 above)
Comment (protatremy) [inline master/docs/developer/secrets.rst]: ok

File: master/docs/developer/secrets.rst (hunk 8 above)
Comment (protatremy) [inline master/docs/developer/secrets.rst]: ok

File: master/docs/developer/secrets.rst (hunk 9 above)
Comment (protatremy) [inline master/docs/developer/secrets.rst]: ok

Comment (protatremy) [inline master/docs/developer/secrets.rst]: thanks a lot for reviewing!

**Pull request merged**